internal/oidc/service_test.go

```diff
@@ -66,3 +74,136 @@ func TestService_Get(t *testing.T) {
 		})
 	}
 }
+
+func TestService_GetForTenant(t *testing.T) {
+	tests := []struct {
+		name         string
+		tenantID     string
+		oidcRepo     *oidcmock.Repository
+		wantProvider oidc.Provider
+		assertErr    assert.ErrorAssertionFunc
+	}{
+		{
+			name:         "Success",
+			tenantID:     tenantID,
+			oidcRepo:     newOIDCRepo(nil, nil, nil, nil, nil),
+			wantProvider: oidcProvider,
+			assertErr:    assert.NoError,
+		},
+		{
+			name:      "Get OIDC error",
+			oidcRepo:  newOIDCRepo(nil, errors.New("Repository.GetForTenant() error"), nil, nil, nil),
+			tenantID:  "does-not-exist",
+			assertErr: assert.Error,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := oidc.NewService(tt.oidcRepo)
+
+			gotProvider, err := s.GetForTenant(t.Context(), tt.tenantID)
+			if !tt.assertErr(t, err, fmt.Sprintf("Service.GetForTenant() error %v", err)) || err != nil {
+				assert.Zerof(t, gotProvider, "Service.GetForTenant() extected zero value if an error is returned, got %v", gotProvider)
+				return
+			}
+
+			assert.Equal(t, tt.wantProvider, gotProvider, "Service.GetForTenant()")
+		})
+	}
+}
+
+func TestService_Create(t *testing.T) {
+	tests := []struct {
+		name      string
+		tenantID  string
+		oidcRepo  *oidcmock.Repository
+		assertErr assert.ErrorAssertionFunc
+	}{
+		{
+			name:      "Success",
+			tenantID:  tenantID,
+			oidcRepo:  newOIDCRepo(nil, nil, nil, nil, nil),
+			assertErr: assert.NoError,
+		},
+		{
+			name:      "Get OIDC error",
+			oidcRepo:  newOIDCRepo(nil, nil, errors.New("Repository.Create() error"), nil, nil),
+			tenantID:  "does-not-exist",
+			assertErr: assert.Error,
+		},
```

Comment:
`Create` method shouldn't return an error if something it tries to create doesn't exist.